recipes/catch2/2.x.x/conanfile.py

```diff
@@ -108,16 +110,20 @@ def package_id(self):
             self.info.header_only()
 
     def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "Catch2")
+        self.cpp_info.set_property("cmake_target_name", "Catch2::Catch2")
```

Comment:
I would put `self.cpp_info.set_property("cmake_target_name", "Catch2::Catch2")` under else branch.

Reply:
This prevents the unofficial `catch2::catch2` (not the lowercase) target from being created at all.

Reply:
there is no other choise when `with_main=True`

Reply:
Putting it in the else branch will make `CMakeDeps` generate `catch2::catch2` as well. Is this a bug in Conan then?

Reply:
you mean even when `catch2:with_main` is False?

Reply:
When this is moved into the else branch:

`with_main=False`:

```cmake
# Catch2-release-x86_64-data.cmake#L4
set(catch2_COMPONENT_NAMES "")
# Catch2Targets.cmake#L18
    add_library(Catch2::Catch2 INTERFACE IMPORTED)
```

Only `Catch2::Catch2` is created.

`with_main=True`:

```cmake
# Catch2-release-x86_64-data.cmake#L4
list(APPEND catch2_COMPONENT_NAMES Catch2::Catch2 Catch2::Catch2WithMain)
# Catch2Targets.cmake#L18
    add_library(catch2::catch2 INTERFACE IMPORTED)
```

`catch2::catch2`, `Catch2::Catch2` and `Catch2::Catch2WithMain` are created.

With the PR, `catch2::catch2` is not created.

Reply:
As I said, the global target should always depend on all component to avoid side effects in config files generated by conan. When `with_main=True`, `Catch2::Catch2` doesn't depend on `Catch2::Catch2WithMain`.

If you really want to not create an unwanted targets, and avoid side effects, here is a workaround

```
self.cpp_info.set_property("cmake_target_name", "Catch2::Catch2{}".format("WithMain" if self.options.with_main else "")
```